### ISSUE TITLE
The performance of adding elements in `CommandBarFlyout.SecondaryCommands` is poor

### ISSUE THREAD

**Issue by hishitetsu** [microsoft/microsoft-ui-xaml#8359]
### Describe the bug

I want to create context menus with `CommandBarFlyout`, but adding elements in `SecondaryCommands` takes much longer than adding elements in `MenuFlyout`.
In my environment, `CommandBarFlyout.SecondaryCommands.Insert()` takes 50ms to 70ms while `MenuFlyout.Items.Insert()` takes less than 5ms.
This makes it difficult to switch to `CommandBarFlyout`.

### Steps to reproduce the bug

1. Create a `CommandBarFlyout` instance and  an `AppBarButton` instance.
2. Insert the `AppBarButton` to `CommandBarFlyout.SecondaryCommands`.
3. It takes long time.

### Expected behavior

Adding elements in `CommandBarFlyout.SecondaryCommands` should have same performance of adding elements in `MenuFlyout`.

### NuGet package version

WinUI 3 - Windows App SDK 1.2.5: 1.2.230313.1

### Windows version

Windows 11 (22H2): Build 22621

**Comment by yaira2** [comment]
I can reproduce this behavior as well

**Comment by zhuxb711** [comment]
Yes, it takes a long time but fetching context menu from Windows takes more than that so this is not obviously, 
### But I also agree that CommandBarFlyout should be faster without questions.

**Comment by 0x5bfa** [comment]
We're getting many requests to make context flyout more swiftly fast.
I don't get why this has been unthouched at all over a year.
I know the team working on many prioritized things but this issue should be significantly noticeable in File Explorer too.

**Comment by michael-hawker** [comment]
@yair100 @0x5bfa mind retrying on the latest 2.0.1 drop, confirming this is still the case, and attaching a minimal repro here? Thanks!

**Comment by michael-hawker** [comment]
Quick benchmark from code-behind in just constructing these elements (MenuFlyout or CommandBarFlyout) and adding to Items or Primary/SecondaryCommands is all equivalent when adding (in 1.2, 1.8, and 2.0), so this then would seem to indicate that it has something to do with the Visual Tree maybe and being actually attached to some UI? (i.e. the underlying collection types/raw code itself doesn't seem to be the source of the issue.)

**Comment by michael-hawker** [comment]
Confirmed still issue in latest, CommandBarFlyout Secondary perf is worse than primary which is worse than MenuFlyout. It only exhibits when attached to Visual Tree.

Attaching the generated benchmark project which exhibits the issue between MenuFlyout vs. CommandBarFlyout: [FlyoutPerfBench.zip](https://github.com/user-attachments/files/27375865/FlyoutPerfBench.zip)

**Comment by 0x5bfa** [comment]
@niels9001 Can you check if this is fixed by https://github.com/microsoft/microsoft-ui-xaml/commit/5c0970f013029ad4e343ff073fc764f1b49088fe ?

**Comment by yair100** [comment]
We replaced the CommandBarFlyout in Files with a custom MenuFlyout and users confirmed a noticeable improvement to performance.
